server/src/server.ts

```diff
@@ -8160,9 +8162,9 @@ Email verified! You can close this tab or hit the back button.
               !_.isUndefined(xid) && !_.isNull(xid)
                 ? getXidStuff(xid, zid)
                 : Promise.resolve();
-            pidPromise = xidUserPromise.then((xidUser: UserType) => {
+            pidPromise = xidUserPromise.then((xidUser: UserType | "noXidRecord") => {
               shouldCreateXidRecord = xidUser === "noXidRecord";
-              if (xidUser && xidUser.uid) {
```

Comment:
Is there any other scenario where we wouldn't have a uid, or is that enforced by the type?

Reply:
Typescript caught this for me. this `xidUser` variable can be either a UserType object **or** the string "noXidRecord". The type is checked as UserType which will have a uid property.